src/components/weather-page/Weather.tsx

```diff
@@ -54,18 +29,11 @@ export function Weather() {
                 <SingleDayDetail data = {selected}/>  
                 
                 <div className="carousel">
-                    <div onClick ={() => setSelected(mondayData)}><SingleDaySummary data = {mondayData}/></div>
-                    <div onClick ={() => setSelected(tuesdayData)}><SingleDaySummary data = {tuesdayData}/></div>
-                    <div onClick ={() => setSelected(tempData)}><SingleDaySummary data = {tempData}/></div>
-                    <div onClick ={() => setSelected(tempData)}><SingleDaySummary data = {tempData}/></div>
-                    <div onClick ={() => setSelected(tempData)}><SingleDaySummary data = {tempData}/></div>
-                    <div onClick ={() => setSelected(tempData)}><SingleDaySummary data = {tempData}/></div>
-                    <div onClick ={() => setSelected(tempData)}><SingleDaySummary data = {tempData}/></div>
-                      
-                </div>
-                      
+                    {weatherData.map(weatherDay  => <div onClick ={() => setSelectedIndex(weatherDay.id)}><SingleDaySummary data = {weatherDay}/></div>)}
```

Comment:
Do we need this div? Can we just do...

```tsx
weatherData.map(weatherDay  => <SingleDaySummary onClick ={() => setSelectedIndex(weatherDay.id)} data = {weatherDay}/>)
```

Reply:
I've left it as is for now because it creates an error to do with the SingleDaySummary object 

Property 'onClick' does not exist on type 'IntrinsicAttributes & SingleDataDetailProps'

Reply:
fair enough - you might need to tell your component how to handle an `onClick` prop.

Fine to leave it for now.